adafruit_ads1x15/ads1x15.py

```diff
@@ -75,6 +94,9 @@ def __init__(
         gain: float = 1,
         data_rate: Optional[int] = None,
         mode: int = Mode.SINGLE,
+        comparator_queue_length: int = 0,
+        comparator_low_threshold: int = 0x8000,
+        comparator_high_threshold: int = 0x7FF0,
```

Comment:
Why is the low threshold default higher than the default high threshold?

Reply:
This is tricky because the values are in 2's complement, but also they are 12-bit values stored in 16-bit registers where the 4 LSBs are 0. Because of this, 0x8000 is the lowest possible value (decimal -32768 in 16-bit or really -2048 in 12-bit) and 0x7FF0 is the highest possible value (decimal 32752 in 16-bit or really 2047 in 12-bit).
I expanded the comments to hopefully explain this better.

Reply:
Ah! I think it'll be clearer to set the defaults to the actual values and have the properties do the conversion and masking.

Reply:
I set the defaults to the actual (12-bit signed) values so that it is easier to read, and moved the 2's complement and bit shift to 16-bit to the setter functions.

Reply:
I tried to copy the handling of the comparator low/high threshold properties just like the other properties, such as data rate. The subclasses handle the lower level chip differences (12 vs 16 bits, default settings, etc) and pass those back to the higher level ads1x15 class.
Also, the ADC data in the analog_in class is handled as a signed 16-bit integer, so I felt it was appropriate to similarly handle the comparator threshold values as signed 16-bit integers as well.